projects/packages/stats-admin/src/class-admin-post-list-column.php

```diff
@@ -0,0 +1,268 @@
+	}
+
+	/**
+	 * A list of NumberFormatters.
+	 *
+	 * @var \NumberFormatter[]
+	 */
+	private $formatter;
+
+	/**
+	 * The constructor.
+	 */
+	public function __construct() {
+		// Add an icon to see stats in WordPress.com for a particular post.
+		add_action( 'admin_print_styles-edit.php', array( $this, 'stats_load_admin_css' ) );
+
+		add_filter( 'manage_posts_columns', array( $this, 'add_stats_post_table' ) );
+		add_filter( 'manage_pages_columns', array( $this, 'add_stats_post_table' ) );
+
+		add_action( 'manage_posts_custom_column', array( $this, 'add_stats_post_table_cell' ), 10, 2 );
+		add_action( 'manage_pages_custom_column', array( $this, 'add_stats_post_table_cell' ), 10, 2 );
+	}
+
+	/**
+	 * Load CSS needed for Stats column width in WP-Admin area.
+	 *
+	 * @since 4.7.0
+	 */
+	public function stats_load_admin_css() {
+		?>
+		<style type="text/css">
+			.fixed .column-stats {
+				width: 5em;
+			}
+		</style>
+		<?php
+	}
+
+	/**
+	 * Set content for cell with link to an entry's stats in Odyssey Stats.
+	 *
+	 * @param string $column  The name of the column to display.
+	 * @param int    $post_id The current post ID.
+	 *
+	 * @since 4.7.0
+	 */
+	public function add_stats_post_table_cell( $column, $post_id ) {
+		if ( 'stats' === $column ) {
+			if ( 'publish' !== get_post_status( $post_id ) ) {
+				printf(
+					'<span aria-hidden="true">—</span><span class="screen-reader-text">%s</span>',
+					esc_html__( 'No stats', 'jetpack-stats-admin' )
+				);
+			} else {
+				// Link to the wp-admin stats page.
+				$stats_post_url = admin_url( sprintf( 'admin.php?page=stats#!/stats/post/%d/%d', $post_id, \Jetpack_Options::get_option( 'id', 0 ) ) );
+				// Unless the user is on a Default style WOA site, in which case link to Calypso.
+				if ( ( new Host() )->is_woa_site() && Stats_Options::get_option( 'enable_odyssey_stats' ) && 'wp-admin' !== get_option( 'wpcom_admin_interface' ) ) {
+					$stats_post_url = Redirect::get_url(
+						'calypso-stats-post',
+						array(
+							'path' => $post_id,
+						)
+					);
+				}
+
+				static $post_views = null;
+
+				/**
+				 * Jetpack_stats_get_post_page_views_for_current_list makes a request with all post ids in the current $wp_query.
+				 * This way, we'll make a single API request instead of making one for each post.
+				 *
+				 * For this reason, we'll cache the result with the static $post_views variable.
+				 */
+				if ( null === $post_views ) {
+					$post_views = $this->get_post_page_views_for_current_list();
+				}
+
+				$views = $post_views[ $post_id ] ?? null;
+
+				$current_locale = get_bloginfo( 'language' );
+
+				$formatted_views = class_exists( '\NumberFormatter' ) ? $this->get_formatter( $current_locale )->format( $views ) : $this->get_fallback_format_to_compact_version( $views );
+
+				?>
+				<a href="<?php echo esc_url( $stats_post_url ); ?>"
+					title="<?php esc_html__( 'View stats for this post', 'jetpack-stats-admin' ); ?>" target="_blank">
```

Comment:
Opening in a new window seems odd, the current simple site implementation doesn't do that.

Reply:
Hmm ok, I'd guess that that's a hangover from the times when stats used to live solely on WordPress.com rather than something we want to double-down on. CC @ederrengifo

Reply:
@dsas just to be clear, should we remove that redirect now or should we wait until we get more feedback from @ederrengifo? :)

Reply:
Sorry, lets just remove the target, I think it's the right thing to do and it's easy enough to add back if required.